File: src/main/webapp/js/app/master/jenis-pengiriman.service.js
/* 
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */

/* 
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
(function() {

	var services = angular.module('je-master-jenis-pengiriman-service', [ 'ngResource' ]);

	
	services.factory('JenisPengirimanListFactory', [ '$resource', function($resource) {

		return $resource('service/master/jenis-pengiriman', {}, {
			query : {
				method : 'GET',
				isArray : false
			},
			create : {
				method : 'POST'
			}
		});

	} ]);
	
	services.factory('JenisPengirimanEditFactory', [ '$resource', function($resource) {

		return $resource('service/master/jenis-pengiriman/:id', {}, {
			show : {
				method : 'GET',
				params: {id: '@id'}
			},
			update : {
				method : 'PUT',
				params :{id: '@id'}
			},
			remove : {
				method : 'DELETE',
				params :{id: '@id'}
			}
		});

	} ]);
	
})();

File: src/main/java/com/je/spring/rest/master/service/JenisPengirimanService.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package com.je.spring.rest.master.service;

import java.util.List;
import com.je.spring.rest.master.model.JenisPengiriman;
/**
 *
 * @author Prasetyo
 */
public interface JenisPengirimanService {
    
    public List<JenisPengiriman> getAll();
    public JenisPengiriman getById(int id);
    public long count();
    public void insert(JenisPengiriman jenisPengiriman);
    public void update(JenisPengiriman jenisPengiriman);
    public void delete(JenisPengiriman jenisPengiriman);
    
}
